qiskit_ibm_provider/utils/backend_converter.py

```diff
@@ -129,13 +134,13 @@ def convert_to_target(
     return target
 
 
-def qubit_props_dict_from_props(
+def qubit_props_list_from_props(
     properties: BackendProperties,
-) -> Dict[int, IBMQubitProperties]:
+) -> List[IBMQubitProperties]:
     """Uses BackendProperties to construct
-    and return IBMQubitProperties class.
+    and return a list of IBMQubitProperties.
     """
-    qubit_props = {}
+    qubit_props: List[IBMQubitProperties] = []
```

Comment:
This is going to cause an `IndexError` because you assign with set item below. This either needs to preallocated as None like: 
```python3
    qubit_props: List[IBMQubitProperties] = [None for _ in range(num_qubits]
```
Or the better choice is probably to change the set item to `append() `. I guess a third option is to use an intermediate dict and return a sorted list from that.

Reply:
Thanks for spotting this! Fixed.